services/spar/src/Spar/Scim.hs

```diff
@@ -373,77 +415,72 @@ toScimEmail (Email eLocal eDomain) =
 -- 2. We want generic error descriptions in response bodies, while still
 --    logging nice error messages internally.
 
-instance SCIM.Class.User.UserDB Spar where
+instance Scim.Class.User.UserDB Spar where
   -- | List all users, possibly filtered by some predicate.
   list :: ScimTokenInfo
-       -> Maybe SCIM.Filter
-       -> SCIM.SCIMHandler Spar (SCIM.ListResponse SCIM.Class.User.StoredUser)
+       -> Maybe Scim.Filter
+       -> Scim.ScimHandler Spar (Scim.ListResponse Scim.Class.User.StoredUser)
   list ScimTokenInfo{stiTeam} mbFilter = do
     members <- lift $ getTeamMembers stiTeam
     brigusers :: [User]
       <- filter (not . userDeleted) <$>
          lift (Intra.Brig.getUsers ((^. Galley.userId) <$> members))
-    scimusers :: [SCIM.Class.User.StoredUser]
+    scimusers :: [Scim.Class.User.StoredUser]
       <- lift . wrapMonadClient . Data.getScimUsers $ Brig.userId <$> brigusers
     let check user = case mbFilter of
           Nothing -> pure True
           Just filter_ ->
             let user' = Scim.Common.value (Scim.thing user)
             in case Scim.filterUser filter_ user' of
                  Right res -> pure res
-                 Left err  -> SCIM.throwSCIM $
-                   SCIM.badRequest SCIM.InvalidFilter (Just err)
+                 Left err  -> Scim.throwScim $
+                   Scim.badRequest Scim.InvalidFilter (Just err)
     -- FUTUREWORK: once bigger teams arrive, we should have pagination here.
-    SCIM.fromList <$> filterM check scimusers
+    Scim.fromList <$> filterM check scimusers
 
   -- | Get a single user by its ID.
   get :: ScimTokenInfo
       -> Text
-      -> SCIM.SCIMHandler Spar (Maybe SCIM.Class.User.StoredUser)
+      -> Scim.ScimHandler Spar (Maybe Scim.Class.User.StoredUser)
   get ScimTokenInfo{stiTeam} uidText = do
     uid <- parseUid uidText
-    briguser <- lift $ Intra.Brig.getUser uid
-    scimuser <- maybe (pure Nothing)
-                      (lift . wrapMonadClient . Data.getScimUser . Brig.userId)
-                      briguser
-    if ( userTeam (fromJust briguser) /= Just stiTeam ||
-         userDeleted (fromJust briguser)
```

Comment:
This started failing when `getUser` started returning `Nothing`. I fixed that and also rewrote the function slightly.